File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/domain/Bilet.cs
namespace AgentiiZboruriC.domain;

using System;

[Serializable]
public class Bilet : Entity<int>
{
    internal Angajat Angajat;
    internal Zbor Zbor;
    internal Persoana Persoana;

    public Bilet(int id, Angajat angajat, Zbor zbor, Persoana persoana) : base(id)
    {
        this.Angajat = angajat;
        this.Zbor = zbor;
        this.Persoana = persoana;
    }

    public Angajat GetAngajat()
    {
        return Angajat;
    }

    public Zbor GetZbor()
    {
        return Zbor;
    }

    public Persoana GetPersoana()
    {
        return Persoana;
    }

    public override string ToString()
    {
        return $"Bilet{{angajat={Angajat}, zbor={Zbor}, persoana={Persoana}}}";
    }

    public void SetAngajat(Angajat angajat)
    {
        this.Angajat = angajat;
    }

    public void SetZbor(Zbor zbor)
    {
        this.Zbor = zbor;
    }

    public void SetPersoana(Persoana persoana)
    {
        this.Persoana = persoana;
    }

    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;

        Bilet bilet = (Bilet)obj;

        return Angajat == bilet.Angajat && Zbor == bilet.Zbor && Persoana == bilet.Persoana;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Angajat, Zbor, Persoana);
    }

    public int GetId()
    {
        return base.GetId();
    }

    public void SetId(int id)
    {
        base.SetId(id);
    }
}

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Model/src/main/java/ro/model/Bilet.java
package ro.model;

public class Bilet extends Entity<Integer>{
    Angajat angajat;
    Zbor zbor;
    Persoana persoana;

    public Bilet(Integer along, Angajat angajat, Zbor zbor, Persoana persoana) {
        super(along);
        this.angajat = angajat;
        this.zbor = zbor;
        this.persoana = persoana;
    }

    public Angajat getAngajat() {
        return angajat;
    }

    public Zbor getZbor() {
        return zbor;
    }

    public Persoana getPersoana() {
        return persoana;
    }

    @Override
    public String toString() {
        return "Bilet{" +
                "angajat=" + angajat +
                ", zbor=" + zbor +
                ", persoana=" + persoana +
                '}';
    }

    public void setAngajat(Angajat angajat) {
        this.angajat = angajat;
    }

    public void setZbor(Zbor zbor) {
        this.zbor = zbor;
    }

    public void setPersoana(Persoana persoana) {
        this.persoana = persoana;
    }

    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Bilet bilet = (Bilet) o;

        if (angajat != null ? !angajat.equals(bilet.angajat) : bilet.angajat != null) return false;
        if (zbor != null ? !zbor.equals(bilet.zbor) : bilet.zbor != null) return false;
        return persoana != null ? persoana.equals(bilet.persoana) : bilet.persoana == null;
    }

    public int hashCode() {
        int result = angajat != null ? angajat.hashCode() : 0;
        result = 31 * result + (zbor != null ? zbor.hashCode() : 0);
        result = 31 * result + (persoana != null ? persoana.hashCode() : 0);
        return result;
    }

    public Integer getId() {
        return super.getId();
    }

    public void setId(Integer id) {
        super.setId(id);
    }

}
